tests/model_fields/models.py

```diff
@@ -469,3 +471,121 @@ class UUIDChild(PrimaryKeyUUIDModel):
 
 class UUIDGrandchild(UUIDChild):
     pass
+
+
+class GeneratedModel(models.Model):
+    a = models.IntegerField()
+    b = models.IntegerField()
+    field = models.GeneratedField(expression=F("a") + F("b"), db_persist=True)
+
+    class Meta:
+        required_db_features = {
+            "supports_generated_columns",
+            "supports_stored_generated_columns",
+        }
+
+
+class GeneratedModelParams(models.Model):
+    field = models.GeneratedField(
+        expression=Value("Constant", output_field=CharField(max_length=10))
+    )
+
+    class Meta:
+        required_db_features = {
+            "supports_generated_columns",
+            "supports_generated_columns_params",
+            "supports_stored_generated_columns",
+        }
+
+
+class GeneratedModelVirtual(models.Model):
+    name = models.CharField(max_length=10)
+    lower_name = models.GeneratedField(expression=Lower("name"), db_persist=False)
+
+    class Meta:
+        required_db_features = {
+            "supports_generated_columns",
+            "supports_virtual_generated_columns",
+        }
+
+
+class GeneratedModeOutputField(models.Model):
+    name = models.CharField(max_length=10)
+    lower_name = models.GeneratedField(
+        expression=Lower("name"),
+        output_field=TextField(
+            db_collation=connection.features.test_collations.get("non_default")
+        ),
+    )
+
+    class Meta:
+        required_db_features = {
+            "supports_generated_columns",
+            "supports_stored_generated_columns",
+        }
+
+
+class GeneratedModeOutputFieldVirtual(models.Model):
+    name = models.CharField(max_length=10)
+    lower_name = models.GeneratedField(
+        expression=Lower("name"),
+        db_persist=False,
+        output_field=CharField(
+            db_collation=connection.features.test_collations.get("non_default"),
+            max_length=11,
```

Comment:
I used `CharField` with `max_length=11` instead of `TextField` so Oracle didn't error in `test_output_field_virtual`.